socorro/external/es/connection_context.py

```diff
@@ -153,64 +58,18 @@ def __call__(self, name=None, timeout=None):
         conn = self.connection(name, timeout)
         yield conn
 
-    def get_socorro_index_settings(self, mappings):
-        """Return a dictionary containing settings for an Elasticsearch index."""
-        return {
-            "settings": {
-                "index": {
-                    "number_of_shards": self.config.elasticsearch_shards_per_index,
-                    "query": ES_QUERY_SETTINGS,
-                    "analysis": ES_CUSTOM_ANALYZERS,
-                }
-            },
-            "mappings": mappings,
-        }
-
-    def get_mapping(self, index_name, es_doctype, reraise=False):
-        """Retrieves the mapping for a given index and doctype
-
-        NOTE(willkg): Mappings are cached on the ConnectionContext instance. If you
-        change the indices (like in tests), you should get a new ConnectionContext
-        instance.
-
-        :arg str index_name: the index to retrieve the mapping for
-        :arg str es_doctype: the doctype to retrieve the mapping for
-        :arg bool reraise: True if you want this to reraise a NotFoundError; False
-            otherwise
-
-        :returns: mapping as a dict or None
-
-        """
-        cache_key = f"{index_name}::{es_doctype}"
-        mapping = self._mapping_cache.get(cache_key)
-        if mapping is None:
-            try:
-                resp = self.indices_client().get_mapping(index=index_name)
-                mapping = resp[index_name]["mappings"][es_doctype]["properties"]
-                self._mapping_cache[cache_key] = mapping
-            except elasticsearch.exceptions.NotFoundError:
-                if reraise:
-                    raise
-        return mapping
-
-    def create_index(self, index_name, mappings=None):
+    def create_index(self, index_name, index_settings):
         """Create an index that will receive crash reports.
 
         :arg index_name: the name of the index to create
-        :arg mappings: dict of doctype->ES mapping
+        :arg index_settings: index settings including the mapping
 
-        :returns: True if the index was created, False if it already
-            existed
+        :returns: True if the index was created, False if it already existed
```

Comment:
This is sort of a lie. It's True if it was created and False if it already existed or there was some other error. I'm not going to fix this now, but I will add a FIXME.

Reply:
My bad--the relevant code is obscured by the lack of diff context. Scratch this.